### PR TITLE
DOC: capitalize Python as proper noun

### PR DESCRIPTION
**Pull request by erictleung**
- [ ] closes #xxxx
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by MarcoGorelli** [comment]
Thanks - can you also fix this in other doc pages, such as doc/source/ecosystem.rst and doc/source/user_guide/cookbook.rst? There's a few other places where it appears uncapitalised

File: doc/source/user_guide/cookbook.rst [hunk 1]
```diff
@@ -18,7 +18,7 @@ above what the in-line examples offer.
 pandas (pd) and Numpy (np) are the only two abbreviated imported modules. The rest are kept
 explicitly imported for newer users.
 
-These examples are written for Python 3.  Minor tweaks might be necessary for earlier python
+These examples are written for Python 3.  Minor tweaks might be necessary for earlier Python
 versions.
```
Comment (simonjayhawkins) [inline doc/source/user_guide/cookbook.rst:22]: pandas is Python 3 only. Maybe this paragraph could be removed?

**Comment by erictleung** [comment]
> Thanks - can you also fix this in other doc pages, such as doc/source/ecosystem.rst and doc/source/user_guide/cookbook.rst? There's a few other places where it appears uncapitalised

@MarcoGorelli sure thing! I ran:

```
grep python -r *
grep jupyter -r *
```

in the `doc/source/` directory to find instances and changed them. I also found some other random capitalizations along the way.

I didn't touch anything in the `whatsnew/` directory.

File: doc/source/getting_started/intro_tutorials/04_plotting.rst [hunk 2]
```diff
@@ -131,8 +131,8 @@ standard Python to get an overview of the available plot methods:
     ]
 
 .. note::
-    In many development environments as well as ipython and
-    jupyter notebook, use the TAB button to get an overview of the available
+    In many development environments as well as IPython and
+    Jupyter notebook, use the TAB button to get an overview of the available
```
Comment (MarcoGorelli) [inline doc/source/getting_started/intro_tutorials/04_plotting.rst:135]: looks like this should be capitalised too 
```suggestion
    Jupyter Notebook, use the TAB button to get an overview of the available
```
and `numpy` should be `NumPy` https://numpy.org/ . Rest looks good!

**Comment by erictleung** [comment]
@MarcoGorelli hopefully got them now! Thanks for the suggestions. (Side note, just realized I can't spell my branches right and caused me some confusion while making edits 😅)

**Review by MarcoGorelli — APPROVED** [review]
cool, thanks!

**Comment by jreback** [comment]
thanks @erictleung

**Pull request merged**